### PR TITLE
Improve Nightly Test Debugging and Reliability with Docker Uploads

### PR DESCRIPTION
**Pull request by t-eckert**
Changes proposed in this PR:
- Add checks to verify deletion of resources in Destroy method on Helm Cluster

How I've tested this PR:
I set up this PR run nightly-acceptance-tests.

How I expect reviewers to test this PR:
Watch the nightly-acceptance-tests and see if they succeed.

Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: control-plane/Makefile [hunk 1]
```diff
@@ -46,6 +46,25 @@ ifeq ($(CIRCLE_BRANCH), main)
 	@docker push $(CI_DEV_DOCKER_NAMESPACE)/$(CI_DEV_DOCKER_IMAGE_NAME):latest
 endif
 
+# TODO: Remove this ci.nightly-dev-docker target once we move the acceptance tests to Github Actions.
+# In CircleCI, the linux binary will be attached from a previous step at pkg/bin/linux_amd64/. This make target
+# should only run in CI and not locally.
```
Comment (ndhanushkodi) [inline control-plane/Makefile:51]:
```suggestion
# should only run in CI and not locally. This is a guardrail for a case where the latest commit on main does not succeed to upload a docker image, so with this step the nightly test will upload its own docker image off of main before starting.
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -891,6 +905,23 @@ workflows:
       - acceptance-tproxy:
           requires:
             - dev-upload-docker
+      - cleanup-gcp-resources
```
Comment (ndhanushkodi) [inline .circleci/config.yml:908]: reminder to remove 908-924 before merging :)

**Review by ndhanushkodi — APPROVED** [review]
These look good Thomas! They'll help us realize the underlying issues a little sooner rather than having to dig for them. Thanks!!

**Review by thisisnotashwin — APPROVED** [review]
This looks good! Could you update the commit to say debug nightly tests? I don't know if this strictly fixes our tests.

**Comment by t-eckert** [comment]
> Could you update the commit to say debug nightly tests?

Is what I changed it to good? I didn't update the commit, but when it squashes it will have the title of the PR as the commit.

I'm OOO today so if you think it's good, go ahead and squash it into `main`.

**Pull request merged**